languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -9966,6 +9966,19 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rule id= "JE_M_EXCUSE " name="Je m’excuse" default="temp_off">
+          <pattern>
+              <token postag="SENT_START"/>
+              <marker>
+                  <token inflected="yes">Je</token>
+                  <token regexp="yes">m'</token>
+                  <token regexp="yes">excuse</token>
+              </marker>
+          </pattern>
+          <message>S'excuser soi-même peut être perçu comme un manque de politesse.Souhaitez-vous écrire <suggestion>Excusez-moi</suggestion> employé dans le langage courant ou           <suggestion>Veuillez m'excuser</suggestion>
+              employé dans le langage formel ?</message>
```

Comment:
* Missing space after dot in message.

* The "regexp="yes" are not needed in tokens (they are not regexp).

* we should add a `<url>` to help justify the rule and so that user can learn why.  This one looks good: `<url>https://fr.wiktionary.org/wiki/je_m%E2%80%99excuse</url>`

* I don't think CAT_GRAMMAIRE" is right.  "Je m’excuse" is grammatically correct, but some people think it's more polite to say "Excusez-moi". How about category "CAT_TOURS_CRITIQUE"?

* I think the rule should have `type="style"`